pennylane/operation.py

```diff
@@ -1782,6 +1786,8 @@ def __mul__(self, a):
         r"""The scalar multiplication operation between a scalar and an Observable/Tensor."""
         if isinstance(a, (int, float)):
             return qml.Hamiltonian([a], [self], simplify=True)
+        if inspect.isfunction(a):
+            return qml.ops.ParametrizedHamiltonian([a], [self])  # pylint: disable=no-member
```

Comment:
I guess we would need to if we want to allow using any of the `SymbolicOp` or `CompositeOp` classes, since you could define observables there that are not `Observables`. Do we then want to add any checks instead of this one (f.x. regarding hermiticity), or do we just put it 100% in the hands of the user to ensure that they are creating a valid Hamiltonian?

Reply:
My understanding was that hermiticity checks are too expensive and sometimes misleading (since a Hermitian operator can be composed of non-Hermitian operators). So I would not include any such checks here either

Reply:
Okay, sound good - I'll move the `__mul__` logic to operator and remove the Observables check!